leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -312,20 +321,100 @@ public static void main(final String[] args) {
                 return;
             }
         }
+
         // Set boolean controlling OSCORE usage
-        boolean useOSCore = false;
+        OSCoreSettings oscoreSettings = null;
         if (cl.hasOption("oscore")) {
             System.out.println("Using OSCORE");
-            useOSCore = true;
 
             HashMapCtxDB db = OscoreHandler.getContextDB();
             OSCoreCoapStackFactory.useAsDefault(db);
 
+            // Parse OSCORE related command line parameters
+
+            String mastersecretStr = cl.getOptionValue("mastersecret");
+            String mastersaltStr = cl.getOptionValue("mastersalt");
+            String idcontextStr = cl.getOptionValue("idcontext");
+            String senderidStr = cl.getOptionValue("senderid");
+            String recipientidStr = cl.getOptionValue("recipientid");
+            String aeadStr = cl.getOptionValue("aead");
+            String hkdfStr = cl.getOptionValue("hkdf");
```

Comment:
could you move the `String param = cl.getOptionValue("param") near their check ?

Reply:
Done in commit 5d2aa3943fa4944954b1a2399f39a65a54da7e1e